Configuration/src/main/java/io/deephaven/configuration/ParsedProperties.java

```diff
@@ -105,7 +105,17 @@ public ParsedProperties() {
      *        PropertyInspector when checking whether required or disallowed properties are present.
      */
     public ParsedProperties(final boolean ignoreScopes) {
-        context = new ConfigurationContext();
+        this(ignoreScopes, new DefaultConfigurationContext());
+    }
+
+    /**
+     * A constructor that starts with no existing scoped or final properties.
+     *
+     * @param ignoreScopes True if this parser should ignore scope restrictions, false otherwise. Used by the
+     *        PropertyInspector when checking whether required or disallowed properties are present.
+     */
+    public ParsedProperties(final boolean ignoreScopes, final @NotNull ConfigurationContext context) {
```

Comment:
`final @NotNull` is backwards from our usual convention. Is there something I need to re-evaluate?

Reply:
Let's keep following the convention we have been in this repo for now. I'd prefer to not change it until/unless we decide to do a style formatting pass over all the files.
I think the only time it materially matters is for array types, anyway, right?